project-novica/demo/src/main/java/com/example/demo/service/AuthenticationService.java

```diff
@@ -0,0 +1,34 @@
+package com.example.demo.service;
+
+import com.example.demo.model.User;
+import com.example.demo.model.exceptions.EmailAlreadyExistException;
+import com.example.demo.model.exceptions.InvalidArgumentException;
+
+
+import javax.persistence.criteria.CriteriaBuilder;
+import java.util.List;
+
+public interface AuthenticationService {
+    //Create Read Update Delete
+    boolean checkUserName(String firstName);
```

Comment:
Refactor this method, so that every method in the authentication service is only used for authentication purposes